lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -1015,6 +1015,42 @@ def _order_problems(blocks):
     return problems
 
 
+def parse_student_data(student_data):
+    """
+    Parses student data to output a more easily-readable version of the 'state', which includes last submission time,
+    number of attempts and the students' answers for a specific problem. We have 2 cases:
+    1) for students with field 'student_answers', we output all the fields as they are (maintaining JSON format if
+        applicable
+    2) for students with field 'student_answer' (note singular), we only output columns for student username and that
+        answer
+
+    :param student_data: dict with 2 keys: 'username' and state'. The value of 'state' is a string version of a dict
+
+    :return: 2-d array, where a row is the username and state data for a student. First row is the headers. Only
+        includes students that have submitted an answer to the problem
+    """
+    header = ['username', 'state']
+    rows = []
+
+    for student in student_data:
+        state = student['state']
+        row = [student['username']]
+        try:
+            student_data_dict = json.loads(state)
+            if 'student_answer' in student_data_dict:
+                row_field = student_data_dict['student_answer']
+                row.append(row_field)
+                rows.append(row)
+            else:
+                row.append(state)
+                rows.append(row)
+        except ValueError:  # if the state is not a valid JSON-formatted string (e.g. 'state1'), append state as-is
```

Comment:
Which line actually throws the `ValueError`?
I suspect we can move much of this out of the `try` block.

Also, it looks like we always append row to rows, no matter the condition,
so that could be pulled out from the branch, and instead be shared.
Even row_field can be abstracted.

I'm guessing this could be as simple as:
```python
try:
    student_data_dict = json.loads(state)
except ValueError:
    row_field = state
else:
    row_field = student_data_dict.get('student_answer', state)
row.append(row_field)
rows.append(row)
```